reliability-v2/start.sh

```diff
@@ -217,6 +219,43 @@ elif [[ ! -z $kubeconfig && ! -z $users_spec && ! -z $kubeadmin_password ]]; the
     KUBECONFIG=$kubeconfig
 fi
 
+# Deploy NFS storage class for cluster without storageclass
+cd $RELIABILITY_DIR
+
+if [[ $(oc get storageclass -o json | jq .items) == "[]" ]];then
+    cd utils
+    echo "Deploy nfs-provisioner"
+    #wget --quiet https://gitlab.cee.redhat.com/-/ide/project/wduan/openshift_storage/tree/master/-/nfs/deploy_nfs_provisioner.sh -O 
+    #chmod +x deploy_nfs_provisioner.sh
+    ./deploy_nfs_provisioner.sh
+    cd -
+fi
+
+# Configure storage for monitoring
+# https://docs.openshift.com/container-platform/4.12/scalability_and_performance/scaling-cluster-monitoring-operator.html#configuring-cluster-monitoring_cluster-monitoring-operator
+oc get cm -n openshift-monitoring cluster-monitoring-config -o yaml  | grep volumeClaimTemplate > /dev/null 2>&1
+if [[ $? -eq 1  ]]; then
+    export STORAGE_CLASS=$(oc get storageclass | grep default | awk '{print $1}')
+    export PROMETHEUS_RETENTION_PERIOD=30d
+    export PROMETHEUS_STORAGE_SIZE=500Gi
+    export ALERTMANAGER_STORAGE_SIZE=20Gi
+    envsubst < content/cluster-monitoring-config.yaml | oc apply -f -
+    sleep 60
+    oc rollout status -n openshift-monitoring deploy/cluster-monitoring-operator
+    oc rollout status -n openshift-monitoring sts/prometheus-k8s
+    token=$(oc create token -n openshift-monitoring prometheus-k8s --duration=6h)
+    URL=https://$(oc get route -n openshift-monitoring prometheus-k8s -o jsonpath="{.spec.host}")
+    prom_status="not_started"
+    sleep 30
+    while [[ "$prom_status" != "success" ]]; do
+        prom_status=$(curl -s -g -k -X GET -H "Authorization: Bearer $token" -H 'Accept: application/json' -H 'Content-Type: application/json' "$URL/api/v1/query?query=up" | jq -r '.status')
+        sleep 5
+        echo "Prometheus status not ready yet, retrying in 5s..."
```

Comment:
@qiliRedHat In case prometheus is taking forever to be ready - instead of infinite loop until prometheus is ready, how about a timeout and gracefully exiting with the reason for the failure why prometheus is not ready?

Reply:
Looks good to me otherwise!

Reply:
@svetsa-rh Thank you!
I added what you suggested.
Test result:
```
Sleep 30s to wait for prometheus status to become success.
Prometheus status is not success yet, retrying in 10s, retries left: 19.
Prometheus is success now.
```
Yes this is similar to post config, but does't use INFRA node - to save resource in a long run. Thanks for the info about that issue. I haven't see this in my recent runs with this branch yet. I'll keep an eye on it.

Reply:
@svetsa-rh PTAL again, thanks